lib/internal/timers.js

```diff
@@ -391,13 +391,11 @@ function getTimerDuration(msecs, name) {
 
 function compareTimersLists(a, b) {
   const expiryDiff = a.expiry - b.expiry;
-  if (expiryDiff === 0) {
-    if (a.id < b.id)
-      return -1;
-    if (a.id > b.id)
-      return 1;
-  }
-  return expiryDiff;
+  if (expiryDiff !== 0)
+    return expiryDiff;
+  if (a.id < b.id)
+    return -1;
+  return 1;
```

Comment:
This isn't correct, as-is this will return `1` if both `id`s are the same.

Reply:
As we are incrementing the id's (unable to find a decrement block for timeout id). so I think there won't be a case of the same ids. However, I think I need some research around this and will open a fresh PR then.